test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -540,6 +554,45 @@
                  (initialize-master-ssl! settings "master" ca-settings 512)))
             (fs/copy copy path)))))))
 
+(deftest initialize-master-ssl!-test-with-keylength-settings
+  (let [tmp-confdir (fs/copy-dir confdir (ks/temp-dir))
+        settings (-> (testutils/master-settings tmp-confdir "master")
+                     (assoc :dns-alt-names "onefish,twofish"))
+        ca-settings (assoc (testutils/ca-settings (str tmp-confdir "/ssl/ca")) :keylength 768)]
+
+    (retrieve-ca-cert! (:cacert ca-settings) (:localcacert settings))
+    (initialize-master-ssl! settings "master" ca-settings)
+
+    (testing "hostprivkey should have correct keylength"
+      (let [key (-> settings :hostprivkey utils/pem->private-key)]
+        (is (utils/private-key? key))
+        (is (= 768 (utils/keylength key)))))
+
+    (testing "hostpubkey should have correct keylength"
+      (let [key (-> settings :hostpubkey utils/pem->public-key)]
+        (is (utils/public-key? key))
+        (is (= 768 (utils/keylength key)))))))
+
+(deftest initialize-master-ssl!-test-with-incorrect-keylength
+  (let [tmp-confdir (fs/copy-dir confdir (ks/temp-dir))
+        settings (-> (testutils/master-settings tmp-confdir "master")
+                     (assoc :dns-alt-names "onefish,twofish"))
+        ca-settings (testutils/ca-settings (str tmp-confdir "/ssl/ca"))]
+
+    (retrieve-ca-cert! (:cacert ca-settings) (:localcacert settings))
+
+    (testing "should throw an error message with too short keylength"
+      (is (thrown-with-msg?
+            java.security.InvalidParameterException
+            #".*RSA keys must be at least 512 bits long.*"
+            (initialize-master-ssl! settings "master" (assoc ca-settings :keylength 128)))))
```

Comment:
Regarding my general comment about putting keylength into the settings maps...if we did that (and it was in *both* CA and master settings) then we would move the assoc over to the `settings` since that's the one that the master would be getting it from in `initialize-master-ssl!`.